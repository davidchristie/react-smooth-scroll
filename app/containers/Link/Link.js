import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import { scrollToElementId } from '../../actions/scroll'
import styles from './Link.css'

class Link extends React.Component {
  render () {
    const classes = [styles.link]
    if (this.props.isActive) classes.push(styles.active)
    return (
      <span className={classes.join(' ')} onClick={this.props.onClick}>
        {this.props.children}
      </span>
    )
  }
}

Link.propTypes = {
  to: PropTypes.string.isRequired
}

export default connect(
  (state, ownProps) => {
    return {
      isActive: state.scroll.targetId === ownProps.to
    }
  },
  (dispatch, ownProps) => {
    return {
      onClick: event => dispatch(scrollToElementId(ownProps.to))
    }
  }
)(Link)
