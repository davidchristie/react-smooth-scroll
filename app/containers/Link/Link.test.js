import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import Link from '../Link'

const mockStore = configureStore([thunk])

describe('<Link />', () => {
  it('should match snapshot', () => {
    const initialState = {
      scroll: {}
    }
    const store = mockStore(initialState)
    const wrapper = mount(
      <Provider store={store}>
        <Link to='elementId' />
      </Provider>
    )
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
