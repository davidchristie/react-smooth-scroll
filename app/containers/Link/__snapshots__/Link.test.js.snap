// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Link)
    to="elementId"
  >
    <Link
      isActive={false}
      onClick={[Function]}
      to="elementId"
    >
      <span
        className="link"
        onClick={[Function]}
      />
    </Link>
  </Connect(Link)>
</Provider>
`;
